src/modules/cpu/host_fused_functions.hpp

```diff
@@ -4,12 +4,593 @@
+        // __m128 pTransposedCtmRow0 = _mm_load_ps(ctm); // Load ctm row 0
+        // __m128 pTransposedCtmRow1 = _mm_load_ps(ctm + 4); // Load ctm row 1
+        // __m128 pTransposedCtmRow2 = _mm_load_ps(ctm + 8); // Load ctm row 2
+        // __m128 pTransposedCtmRow3 = _mm_load_ps(ctm + 12); // Load ctm row 3
+
+        // _MM_TRANSPOSE4_PS(pTransposedCtmRow0, pTransposedCtmRow1, pTransposedCtmRow2, pTransposedCtmRow3);  // Transpose ctm to contain cols in pTransposedCtmRow0/pTransposedCtmRow1/pTransposedCtmRow2/pTransposedCtmRow3
+
+        // for (int i = 0; i < srcSize.height; i++)
+        // {
+        //     int vectorLoopCount = 0;
+
+        //     for (; vectorLoopCount < alignedLength; vectorLoopCount+=12)
+        //     {
+        //         // To compute matmul4x4 of srcPtr (pixel0/pixel1/pixel2/pixel3) and ctmTransposed (pTransposedCtmRow0/pTransposedCtmRow1/pTransposedCtmRow2/pTransposedCtmRow3)
+        //         //
+        //         //                  --                  --                                                                    --                  --
+        //         //                 | [0],  [1],  [2], [X] |  (R/G/B/X values for pixel n)                                    | [0], [4],  [8], [12] | (stored in pTransposedCtmRow0)
+        //         // srcPtrIndices = | [3],  [4],  [5], [X] |  (R/G/B/X values for pixel n + 1)      X      ctmMatrixIndices = | [1], [5],  [9], [13] | (stored in pTransposedCtmRow1)
+        //         //                 | [6],  [7],  [8], [X] |  (R/G/B/X values for pixel n + 2)                                | [2], [6], [10], [14] | (stored in pTransposedCtmRow2)
+        //         //                 | [9], [10], [11], [X] |  (R/G/B/X values for pixel n + 3)                                | [3], [7], [11], [15] | (stored in pTransposedCtmRow3)
+        //         //                  --                  --                                                                    --                  --
+        //         // This matrix multiplication gives results for 4 pixels in one shot - each row of the result matrix containing one set of R/G/B/X values.
+
+        //         // Load
+
+        //         px0 =  _mm_loadu_si128((__m128i *)srcPtrTemp);
+
+        //         // Unpack u8 to f32
+
+        //         px1 = _mm_unpackhi_epi8(px0, zero);    // pixels 8-15
+        //         px0 = _mm_unpacklo_epi8(px0, zero);    // pixels 0-7
+        //         p0 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px0, zero));    // pixels 0-3
+        //         p1 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px0, zero));    // pixels 4-7
+        //         p2 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px1, zero));    // pixels 8-11
+        //         p3 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px1, zero));    // pixels 12-15
+
+        //         // Shuffle to broadcast
+
+        //         pSrcF32[0] = _mm_shuffle_ps(p0, p0, _MM_SHUFFLE(0,0,0,0)); // pix0R, pix0R, pix0R, pix0R
```

Comment:
use _mm_shuffle_epi8 with mask and convert_ps after getting the right pixels

Reply:
To be revisited after FusedColorJitter convergence